extensions/wikia/ThemeDesigner/js/ThemeDesigner.js

```diff
@@ -537,13 +574,15 @@
 		set: function (setting, newValue) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Function has a complexity of 14 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)